sound/soc/sof/sof-audio.c

```diff
@@ -310,7 +310,7 @@ sof_unprepare_widgets_in_path(struct snd_sof_dev *sdev, struct snd_soc_dapm_widg
 	struct snd_soc_dapm_path *p;
 
 	/* return if the widget is in use or if it is already unprepared */
-	if (!swidget->prepared || swidget->use_count > 1)
+	if (!swidget->prepared || swidget->use_count > 0)
```

Comment:
Can the use_count go negative?
```c
if (!swidget->prepared || swidget->use_count)
```

Reply:
> Can the use_count go negative?
> 
> ```c
> if (!swidget->prepared || swidget->use_count)
> ```

Ideally, no. But I do see it is negative when I meet IPC errors. And sof_widget_free() will never work at that time.